crates/client/sync/src/fetch/mod.rs

```diff
@@ -112,6 +127,57 @@ pub async fn l2_fetch_task(
     Ok(())
 }
 
+enum SyncStatus {
+    Full(u64),
+    UpTo(u64),
+}
+
+async fn sync_blocks(
```

Comment:
can you add some docs here?

Reply:
im unsure where the parallel catching up vs polling phase are here

Reply:
i think it's SyncStatus that i don't understand

Reply:
`SyncStatus` is just used to express if we have reached the tip of the chain. This is significant since we are actually synchronizing from _two_ chains during warp updates:

1. The warp update sender
2. Whatever feeder gateway we have set with `--gateway-url`

The issue is that `--n-blocks-to-sync` will cause sync to finish early, so we need to be able to know if we have reached the tip of the chain or else we do not display '🥳 The sync process has caught up with the tip of the chain'.

Reply:
`sync_n_blocks` is actually _only_ used during parallel fetching: this includes warp updates and normal parallel sync, and does not include post-sync catch up. So `sync_n_blocks` is not called for the polling phase.

Reply:
This has been fixed. Added documentation for `sync_n_blocks` and `SyncStatus`